src/app/collection-page/collection-page.component.ts

```diff
@@ -79,49 +81,52 @@ export class CollectionPage implements OnInit, OnDestroy {
   }
 
   private routeParamSubscrpt(): void {
-    this.route.params.subscribe((routeParams) => {
-      this.colId = routeParams['colId'];
-      // Old links pass a name into the ID, just use that as a search term instead
-      if (!/^[0-9]+$/.test(this.colId)) {
-        this.http.get('/assets/collection-links.json')
-          .subscribe(data => {
-            let linkObj = data
-            let link = linkObj[this.colId]
-            if (link) {
-              this._router.navigateByUrl(link)
-            } else {
-              this._router.navigate(['/search', this.colId.replace('_', ' ')])
-            }
-          })
-      } else if (this.colId) {
-        this._assets.clearAssets();
-          this.getCollectionInfo(this.colId)
-            .then((data) => {
-              this._assets.queryAll(routeParams, true);
-
-              if (!Object.keys(data).length) {
-                throw new Error('No data!');
-              }
-
-              this.assetCount = data['objCount'];
-              this.colName = data['collectionname'];
-              this.colDescription = data['blurburl'];
-              this.colThumbnail = data['bigimageurl'];
-
-              // Set page title
-              this._title.setSubtitle(this.colName)
-            })
-            .catch((error) => {
-              console.error(error);
-              if (error.status === 401){
-                this.unaffiliatedUser = true
-                this.showAccessDeniedModal = true
+    this.route.params.pipe(
+      map(routeParams => {
+        this.colId = routeParams['colId'];
+        // Old links pass a name into the ID, just use that as a search term instead
+        if (!/^[0-9]+$/.test(this.colId)) {
+          this.http.get('/assets/collection-links.json').pipe(
+            map(data => {
+              let linkObj = data
+              let link = linkObj[this.colId]
+              if (link) {
+                this._router.navigateByUrl(link)
+              } else {
+                this._router.navigate(['/search', this.colId.replace('_', ' ')])
               }
-            });
+            }
+          )).subscribe()
+
```

Comment:
@codiak Particularly, here - Compared with the diff, it looks like the .subscribe() should be outside of the conditionals. I'll test this out.